docs/modules/ROOT/pages/guide/meta-tx.adoc

```diff
@@ -0,0 +1,312 @@
+
+In the example code, the functionality of the https://github.com/OpenZeppelin/workshops/blob/master/25-defender-metatx-api/contracts/SimpleRegistry.sol[`SimpleRegistry` contract, window=_blank] is to take a string and store it. The contract's https://github.com/OpenZeppelin/workshops/blob/master/25-defender-metatx-api/contracts/Registry.sol[meta-transaction implementation, window=_blank] achieves the same result by decoupling the signer from the sender of the transaction.
+
+When comparing the code, note the meta-transaction's use of `_msgSender()` as opposed to the SimpleRegistry's use of `msg.sender`. By extending from `ERC2771Context` and `ERC2771Forwarder`, the contract becomes meta-transaction capable.
+
+NOTE: All OpenZeppelin contracts are compatible with the use of `_msgSender()`.
+
+The second fundamental change between the two contracts is the need for the meta-transaction contract (https://github.com/OpenZeppelin/workshops/blob/master/25-defender-metatx-api/contracts/Registry.sol[Registry, window=_blank]) to specify the address of the trusted forwarder, which in this case is the address of the `ERC2771Forwarder` contract.
+
+[[configure-project]]
+== 1. Configure the project
+
+First, fork the repository and navigate to the directory for this guide. There, install the dependencies with `yarn`:
+
+```
+$ git clone https://github.com/openzeppelin/workshops.git
+$ cd workshops/25-defender-metatx-api/
+$ yarn
+```
+
+Create a `.env` file in the project root and supply an API key and secret from the https://defender.openzeppelin.com/v2/#/manage/api-keys/team/new[API keys page, window=_blank]. A private key will be used for local testing but the Relayer is used for actual contract deployment.
+
+```
+PRIVATE_KEY="0xabc"
+API_KEY="abc"
+API_SECRET="abc"
+```
+
+[[create-relayer]]
+== 2. Create Relayer
+
+Run the Relayer creation script, which will use the Defender API parameters in the `.env` file:
+
+```
+$ yarn create-relay
+```
+
+The Relayer is created using the `defender-sdk` package:
+
+[source,jsx]
+----
+// ...
+const client = new Defender(creds);
+
+// Create Relayer using Defender SDK client.
+const requestParams = {
+  name: 'MetaTxRelayer',
+  network: 'sepolia',
+  minBalance: BigInt(1e17).toString(),
+};
+
+const relayer = await client.relay.create(requestParams);
+// ...
+----
+
+After creating it, the script will fetch the Relayer ID and create an API key and secret set to send transacitons via it. The Relayer ID is automatically stored in the `relayer.json` file, and its API paramteres in the `.env` file.
+
+[[compile-using-hardhat]]
+== 3. Compile the Contract Using Hardhat
+
+Within the `contracts` directory, you can find the both the `SimpleRegistry.sol` and `Registry.sol` contracts. The former contract contains the meta-transaction functionality, as you can see here:
+
+[source,jsx]
+----
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
```

Comment:
Should we use a newer version of solidity?

Reply:
Yup, the contracts are compiled using 0.8.23 because of Contracts 5.0